app/utilities/PersistenceManager.java

```diff
@@ -37,6 +38,19 @@ public static User getUserFromRequest(Http.Request request){
         return User.find.where().eq("email", token.getEmail()).findUnique();
     }
 
+    public static void saveNewDefaultPhoto(Restaurant restaurant,Photo photo) {
+        HashMap<String, String> map = new HashMap<>();
+        map.put("isDefault","1");
+        map.put("restaurant.restaurantId", Long.toString(restaurant.getRestaurantId()));
+
+        Photo oldDefault = (Photo) Photo.find.where().allEq((Map) map).findUnique();
```

Comment:
photos are loaded with the restaurant (debug it and see!) so you just need to filter out default photo and mark it as non-default